home/tests.py

```diff
@@ -1,3 +1,79 @@
+from home.models import Profile, Subject, Question, Tag, Question_Tag
+from django.contrib.auth.models import User
+from django.utils import timezone
+import pytest
+
 
 def test_first():
     assert True
+
+
+@pytest.mark.django_db
+def test_add_tags_to_question():
+    # create test data
+    test_question, test_tags = create_test_data()
+    test_question.add_tags_to_question(test_tags)
+
+    # check if all test_tag in relation with test_question
+    assert list(test_question.tags.values()) == [{'id': 5, 'tag_name': 'test_tag_1'},
+                                                 {'id': 6, 'tag_name': 'test_tag_2'},
+                                                 {'id': 7, 'tag_name': 'test_tag_3'}]
+
+    # check if test_question is the only one in relation with test_tag_2
+    test_tag = Tag.objects.get(tag_name='test_tag_2')
+    assert test_tag.questions.values().count() == 1
+
+    # check if (test_question, test_tag_2) in Question_Tag
+    assert Question_Tag.objects.filter(question=test_question, tag=test_tag).exists()
+
+    # check test_tag_3 wasn't added again
+    test_question.add_tags_to_question(['test_tag_3'])
+    assert test_question.tags.values().count() == 3
+
+    # delete test data
+    delete_test_data(test_question.pk, test_tags)
```

Comment:
you don't need this - all the data that you introduce gets deleted automatically between tests since you use the django_db decorator.